## Dataset explanation

- In the practical you will go from raw reads to
taxonomical characterization of metagenome
assembled genomes.
- In this mock experiment we have a small diverse
synthetic rhizosphere community.
- We expose the plant to fungal chitin and we expect the
plant to respond to this treatment by modulating the
microbial community it associates with in the
rhizosphere.
- Can you find out how the microbial community adapts?

## Jupyter Hub

In this course we use a jupyterhub setup to provide access to a linux environment,
and python and R notebooks. Students and instructors have individual
accounts and can log in to the [jupyterhub](https://www.bioinformatics.nl/biosb_metagenomics).

### Logging in

Every participant has an individual account with username/password based on
first n ame a nd l ast n ame f ollowing a fi xed sc heme. 

Username is <first_name>.<last_name> and password is pw4<first_name>, all in
lowercase. Any spaces an dashes in names are represented with low dashes.

Examples:
- rens_holmer, pw4rens
- marnix_medema, pw4marnix

_In theory this makes it easy to find out another personâ€™s credentials, we trust
the participants to not take advantage of this._

### Basic jupyterhub environment

After logging in, an individual session will be started and you will get
access to a launch screen from where you can start linux terminals, and
R and python notebooks (this might look familiar if you know jupyter lab). In
addition, there is a file browser that you can use to browse your home directory.
If you first start your session your home is empty and there will be nothing to
browse.

### Linux terminal

A new terminal session starts in your home folder. Files in your
home folder can be browsed with the file browser on the left of the screen. A
conda environment with all bioinformatics software requirements is available. 
Data, databases and precomputed results are available in /data.
