# Biosb Computational Metagenomics

This course teaches state-of-the-art computational methods for the analysis of metagenome data. Lectures will be combined with hands-on computer sessions using Linux command line tools, Galaxy and R to practice use of the methods on real data. The course will start with foundational knowledge and skills in experimental design, sequencing technologies, quality control, assembly and binning. Subsequently, we will look at various taxonomic assignment algorithms and the use of metagenomic databases such as MGnify. To compare metagenomic features between samples and conditions, we will study various statistical techniques, including differential abundance/expression analysis, association analysis with metadata and causal inference. Finally, we will explore several levels of functional metagenome annotation, including the identification of biosynthetic gene clusters, virulence factors and phages, and community-level metabolic modeling. We will close the course with a keynote lecture on the use of metatranscriptomics to analyse gene expression patterns in microbiomes. After the course, the slides of the presentations and the practicals will remain available for future reference. Software packages used are freeware.

For more info on the course see [https://www.dtls.nl/courses/computational-metagenomics/](https://www.dtls.nl/courses/computational-metagenomics/)