
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../../data-processing/cave-expedition/">
      
      
        <link rel="next" href="../bgcs/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.11">
    
    
      
        <title>Viromics - Biosb Computational Metagenomics</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.4af4bdda.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  
  
    
  
  
  <meta property="og:type" content="website" />
  <meta property="og:title" content="Biosb Computational Metagenomics - Viromics" />
  <meta property="og:description" content="None" />
  <meta property="og:url" content="None" />
  <meta property="og:image" content="Noneassets/images/banner.png" />
  <meta property="og:image:type" content="image/png" />
  <meta property="og:image:width" content="1200" />
  <meta property="og:image:height" content="630" />

  <link rel="alternate" type="application/rss+xml" title="RSS feed of created content" href="Nonefeed_rss_created.xml">
  <link rel="alternate" type="application/rss+xml" title="RSS feed of updated content" href="Nonefeed_rss_updated.xml">

  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#functional-annotation-viromics" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Biosb Computational Metagenomics" class="md-header__button md-logo" aria-label="Biosb Computational Metagenomics" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Biosb Computational Metagenomics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Viromics
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/mdehollander/biosb-metagenomics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    biosb-metagenomics
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Biosb Computational Metagenomics" class="md-nav__button md-logo" aria-label="Biosb Computational Metagenomics" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    Biosb Computational Metagenomics
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/mdehollander/biosb-metagenomics" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    biosb-metagenomics
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 1: Data processing
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Day 1: Data processing
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data-processing/intro/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Introduction
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data-processing/quality-control/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Quality Control
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data-processing/assembly/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data-processing/binning/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Binning
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../data-processing/cave-expedition/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Cave expedition
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day 3: Functional annotation
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Day 3: Functional annotation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Viromics
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Viromics
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#step-1-initial-virus-discovery" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1 - Initial virus discovery
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-host-contamination-removal-and-quality-check" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2 - Host contamination removal and quality check
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-bacterial-host-assignment" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3 - Bacterial host assignment
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-functional-annotation" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4 - Functional annotation
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bgcs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BGCs
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#step-1-initial-virus-discovery" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1 - Initial virus discovery
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-host-contamination-removal-and-quality-check" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2 - Host contamination removal and quality check
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-bacterial-host-assignment" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3 - Bacterial host assignment
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-functional-annotation" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4 - Functional annotation
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="functional-annotation-viromics">Functional annotation - Viromics<a class="headerlink" href="#functional-annotation-viromics" title="Permanent link">&para;</a></h1>
<p>We will now identify viral sequences from metagenomic data. You will be provided with a set of contigs assembled from infant fecal metagenomic samples (<code>/data/precomputed/virome/contigs.fa</code>). Our goals are to:</p>
<ol>
<li>Determine the <strong>viral origin</strong> of the contigs (and assign their taxonomy).</li>
<li>Estimate the <strong>completeness</strong> of viral genomes.</li>
<li>Predict the potential <strong>bacterial host</strong>.</li>
<li>Functionally <strong>annotate</strong> each identified virus.</li>
</ol>
<p>By the end of this exercise, our objective will be to identify the bacteriophage that satisfies all of the following criteria:  </p>
<blockquote>
<ul>
<li>Belongs to the <strong><em>Caudoviricetes</em></strong> class  </li>
<li>Is predicted to have a <strong>complete genome</strong>  </li>
<li>Is predicted to infect the <strong><em>Clostridium</em></strong> genus  </li>
<li>Has the ability to <strong>integrate</strong> into the bacterial genome  </li>
</ul>
</blockquote>
<p><strong>Note:</strong> For this practical, several tools are required for the viral identification and annotation pipeline:  </p>
<blockquote>
<ul>
<li><strong>geNomad</strong>  </li>
<li><strong>CheckV</strong>  </li>
<li><strong>iPHOP</strong>  </li>
</ul>
</blockquote>
<p>All these tools and the necessary databases have been preinstalled and are ready for you to use.</p>
<h2 id="step-1-initial-virus-discovery">Step 1 - Initial virus discovery<a class="headerlink" href="#step-1-initial-virus-discovery" title="Permanent link">&para;</a></h2>
<p>First, we will run <strong>geNomad</strong> to identify viral sequences in our dataset. We will utilize the <code>--enable-score-calibration</code> option to compute false discovery rates (FDR). For this practical exercise, no specific FDR thresholds will be applied. Further, we will add the <code>--cleanup</code> flag to remove intermediate files and the <code>--disable-find-proviruses</code> option to avoid geNomad performing an initial prunning to remove potential contaminant host regions from proviral sequences (we will do this in the next step with CheckV). </p>
<p><strong>Note:</strong> The following command takes about 2 minutes. You can also continue with the results in <code>/data/precomputed/virome/genomad_results/</code>.
<div class="highlight"><pre><span></span><code>genomad<span class="w"> </span>end-to-end<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--enable-score-calibration<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--disable-find-proviruses<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>contigs.fa<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>geNomad_results<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--cleanup<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>/data/databases/genomad/genomad_db/
</code></pre></div></p>
<p>You can find a description of the geNomad output files <a href="https://portal.nersc.gov/genomad/quickstart.html#understanding-the-outputs">here</a>.</p>
<p>Look into <code>geNomad_results/contigs_summary/contigs_virus_summary.tsv</code>. </p>
<details class="done">
<summary>1. How many of the 100 initial contigs are identified as viral?</summary>
<p>55 contigs.</p>
</details>
<details class="done">
<summary>2. What is the length of the largest viral genome that we have identified?</summary>
<p>297,029 bases (MGV_98132).</p>
</details>
<details class="done">
<summary>3. Regarding the viral taxonomy, which is the most common viral class among identified viruses?</summary>
<p><em>Caudoviricetes</em>.</p>
</details>
<details class="done">
<summary>4. Do we identify any single-stranded DNA virus (ssDNA)?</summary>
<p>Yes, there are 13 ssDNA (belonging to <em>Monodnaviria</em> realm).     </p>
</details>
<p><strong>Note:</strong> A useful resource where you can find all the information about the viral taxonomy is <a href="https://ictv.global/taxonomy">here</a>.  </p>
<h2 id="step-2-host-contamination-removal-and-quality-check">Step 2 - Host contamination removal and quality check<a class="headerlink" href="#step-2-host-contamination-removal-and-quality-check" title="Permanent link">&para;</a></h2>
<p>While geNomad performs well at identifying proviruses from metagenomic data, <strong>CheckV</strong> is specifically designed to identify host-virus boundaries with high precision. Further, CheckV allows to estimate the completeness of the predicted viral genomes based on their comparison to a database of complete viral genomes. Therefore, here we use CheckV to quality control the geNomad results and also to trim potential host regions left at the ends of proviruses. </p>
<p><strong>Note:</strong> This command will take about 10 minutes to complete. Again, you can continue with the results in <code>/data/precomputed/virome/checkv_results/</code>.
<div class="highlight"><pre><span></span><code>checkv<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>end_to_end<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>geNomad_results/contigs_summary/contigs_virus.fna<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>CheckV_results<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>-d<span class="w"> </span>/data/databases/checkv-db-v1.4
</code></pre></div></p>
<p>The CheckV output is described <a href="https://bitbucket.org/berkeleylab/checkv/src/master/#markdown-header-output-files">here</a>. Look into <code>CheckV_results/quality_summary.tsv</code>.</p>
<details class="done">
<summary>1. How many proviruses do you find and how many viruses?</summary>
<p>9 proviruses, 46 viruses.</p>
</details>
<details class="done">
<summary>2. How many low, medium, and high quality viruses do we detect? How many complete viruses? How many of them have direct terminal repeats?</summary>
<p>11 viruses with low-quality, 20 with medium-quality and 9 with high-quality.
15 viruses are complete. From them, 10 have direct terminal repeats (DTRs).</p>
</details>
<p><strong>Note:</strong> Depending on the project's goals, it may be advisable to select only viruses predicted to be of at least medium quality. However, for this exercise, we will proceed with all available sequences.</p>
<p><strong>Note:</strong> CheckV provides 2 separate files with the identified proviral and viral sequences. As we want to work with both of them, we combine them into a single file: </p>
<pre><code>cat CheckV_results/proviruses.fna CheckV_results/viruses.fna &gt; CheckV_results/combined.fna
</code></pre>
<h2 id="step-3-bacterial-host-assignment">Step 3 - Bacterial host assignment<a class="headerlink" href="#step-3-bacterial-host-assignment" title="Permanent link">&para;</a></h2>
<p>We will use <strong>iPHoP</strong> for bacterial host assignment of the viruses. Although iPHoP provides both genus- and species-level host predictions, we will focus solely on genus-level assignments. This is because iPHoP offers high-confidence predictions at the genus level (with an estimated false discovery rate of less than 10%), while the confidence decreases at the species level.</p>
<p><strong>Note:</strong> The following command can take around 1 hour. Therefore, you should continue with the results in <code>/data/precomputed/virome/iphop_results/</code>.
<div class="highlight"><pre><span></span><code>mkdir<span class="w"> </span>iphop_results

iphop<span class="w"> </span>predict<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--fa_file<span class="w"> </span>CheckV_results/combined.fna<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--db_dir<span class="w"> </span>/data/databases/Aug_2023_pub_rw/<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--num_threads<span class="w"> </span><span class="m">8</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--out_dir<span class="w"> </span>iphop_results
</code></pre></div></p>
<p>The iPHoP output is described <a href="https://bitbucket.org/srouxjgi/iphop/src/main/#markdown-header-main-output-files">here</a>. </p>
<p><strong>Note:</strong> iPHoP allows to enrich the default database with custom MAGs to improve the host assignment of the viruses.</p>
<p>Look into <code>iphop_results/Host_prediction_to_genus_m90.csv</code>. By default, all virus-host pairs for which the confidence score is higher than the selected cutoff (default = 90) are included. For this exercise, consider only the top hit for each virus (prediction with highest confidence score):</p>
<details class="done">
<summary>1. Is there any virus predicted to infect the <strong><em>Clostridium</em></strong> genus?</summary>
<p>Yes, 2 viruses are predicted to infect <strong><em>Clostridium</em></strong> (CS_426 and CS_2284).</p>
</details>
<details class="done">
<summary>2. Considering the genome completeness estimated in the previous step, which of these <strong><em>Clostridium</em></strong> phages is predicted to have a complete genome?</summary>
<p>Both phages have a complete genome (with DTRs).</p>
</details>
<h2 id="step-4-functional-annotation">Step 4 - Functional annotation<a class="headerlink" href="#step-4-functional-annotation" title="Permanent link">&para;</a></h2>
<p>Multiple tools can be used for the functional annotation of viral genomes. Here, we will use <strong>geNomad</strong> (annotate module) to retrieve the annotations for each of the proteins in our predicted viral genomes.</p>
<p><strong>Note:</strong> The following command takes about 2 minutes. You can also continue with the results in <code>/data/precomputed/virome/genomad_annotation_results/</code>.
<div class="highlight"><pre><span></span><code>genomad<span class="w"> </span>annotate<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>CheckV_results/combined.fna<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>geNomad_annotation_results<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--cleanup<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>/data/databases/genomad/genomad_db/
</code></pre></div></p>
<p>The detailed explanation of this step can be found <a href="https://portal.nersc.gov/genomad/pipeline.html#annotate">here</a>. Check now the <code>geNomad_annotation_results/combined_annotate/combined_genes.tsv</code> file:</p>
<details class="done">
<summary>1. Does any of the identified viruses use an alternative genetic code?</summary>
<p>No, all the viruses use the standard genetic code (translation table 11).</p>
</details>
<details class="done">
<summary>2. Are any of the identified viruses encoding proteins that enable integration into the bacterial genome (integrases)?</summary>
<p>Yes, at least 6 viruses encode integrases (GPD_79092, ELGV_14024, CS_100, CS_659, CS_1726 and CS_2284)  </p>
</details>
<p><strong>Note:</strong> Genetic code 11 (translation table 11) is the standard code used for Bacteria, Archaea, prokaryotic viruses and chloroplast proteins.</p>
<details class="done">
<summary>Which bacteriophage are we looking for?</summary>
<p><strong>CS_2284</strong>: A predicted complete <em>Clostridium</em> phage, belonging to the <em>Caudoviricetes</em> class, that can integrate into the host genome.</p>
</details>







  
    
  
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"/></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date" title="April 2, 2025 14:38:26">April 2, 2025</span>
  </span>

    
    
    
      
  
  <span class="md-source-file__fact">
    <span class="md-icon" title="Contributors">
      
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 4a4 4 0 0 1 4 4 4 4 0 0 1-4 4 4 4 0 0 1-4-4 4 4 0 0 1 4-4m0 10c4.42 0 8 1.79 8 4v2H4v-2c0-2.21 3.58-4 8-4"/></svg>
      
    </span>
    <nav>
      
        <a href="mailto:m.dehollander@nioo.knaw.nl">Mattias de Hollander</a>
    </nav>
  </span>

    
    
  </aside>





                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.expand"], "search": "../../assets/javascripts/workers/search.f8cc74c7.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
  
      <script src="../../assets/javascripts/bundle.c8b220af.min.js"></script>
      
    

  </body>
</html>